server/src/main/java/org/opensearch/indices/recovery/RecoverySettings.java

```diff
@@ -305,6 +325,17 @@ private void setMaxBytesPerSec(ByteSizeValue maxBytesPerSec) {
         }
     }
 
+    private void setSegrepMaxBytesPerSec(ByteSizeValue segrepMaxBytesPerSec) {
```

Comment:
How do we ensure it honours recovery max bytes per sec and doesn't go too off so as to throttle the disc

Reply:
> How do we ensure it honours recovery max bytes per sec and doesn't go too off so as to throttle the disc

Sorry, I just closed this PR by mistake......   Could you reopen it? I'll resolve it later.

Reply:
> How do we ensure it honours recovery max bytes per sec and doesn't go too off so as to throttle the disc

I think it is determined by specific users' cluster. In my own case, if I turn up speed limit to infinity,
The bottleneck of segrep will be in network bandwidth  rather than disk. So leave the decision of this setting to specific user. Of course, it may be better to set the default value to a more conservative value, like 40MB, the same as recovery.

Reply:
changed the setting name to `indices.replication.max_bytes_per_sec`.